types/lodash/lodash-tests.ts

```diff
@@ -7389,3 +7407,152 @@ _.templateSettings; // $ExpectType TemplateSettings
+        type H = GetIndexedField<O1, undefined>; // $ExpectType undefined
+    }
+}
+
+// GetIndexedField: indexing strings
+
+{
+    type S1 = string;
+    type S2 = string & { x: 'OK' };
+
+    type A = GetIndexedField<S1, number>; // $ExpectType string
+    type B = GetIndexedField<S1, 1>; // $ExpectType string
+    type C = GetIndexedField<S1, '1'>; // $ExpectType string
+    type D = GetIndexedField<S1, 'length'>; // $ExpectType number
+    type F = GetIndexedField<S1, 'x'>; // $ExpectType undefined
+    type G = GetIndexedField<S2, 'x'>; // $ExpectType "OK"
+}
+
+// GetIndexedFields: indexing tuples
+
+{
+    type T1 = ['OK'];
+
+    type A = GetIndexedField<T1, number>; // $ExpectType "OK"
+    type B = GetIndexedField<T1, 0>; // $ExpectType "OK"
+    type C = GetIndexedField<T1, 1>; // $ExpectType undefined
+    type D = GetIndexedField<T1, '0'>; // $ExpectType "OK"
+    type E = GetIndexedField<T1, '1'>; // $ExpectType undefined
+    type F = GetIndexedField<T1, 'length'>; // $ExpectType 1
+}
+
+// GetFieldType: correctly handles any possible combination of dots and square
+// brackets in path type literals.
+{
+    interface O1 { key: 'value' };
+    interface O2 { some: { test: { path: 'value' } } };
+
+    interface O3 {
+        'some.key': 'value1';
+        'key[with]brackets': 'value2';
+    };
+
+    type A = GetFieldType<O1, 'key'>; // $ExpectType "value"
+    type B = GetFieldType<O1, '[key]'>; // $ExpectType "value"
+
+    type C = GetFieldType<O2, 'some.test.path'>; // $ExpectType "value"
+    type D = GetFieldType<O2, 'some.test[path]'>; // $ExpectType "value"
+    type E = GetFieldType<O2, 'some[test].path'>; // $ExpectType "value"
+    type F = GetFieldType<O2, '[some].test.path'>; // $ExpectType "value"
+    type G = GetFieldType<O2, 'some[test][path]'>; // $ExpectType "value"
+    type H = GetFieldType<O2, '[some].test[path]'>; // $ExpectType "value"
+    type I = GetFieldType<O2, '[some][test].path'>; // $ExpectType "value"
+    type J = GetFieldType<O2, '[some][test][path]'>; // $ExpectType "value"
+
+    // Here are tricky cases, where dots and square brackets inside key names
+    // mess-up with TypeScript parsing of path type literals we can do.
+
+    // No way to distinguish a single key with dot vs. two dot-separated keys,
+    // GetFieldType<> interpretes it as the later, thus undefined result.
+    type K = GetFieldType<O3, 'some.key'>; // $ExpectType undefined
+
+    // However, wrapping it inside square brackets is a viable work-around
+    // we can support.
+    type L = GetFieldType<O3, '[some.key]'>; // $ExpectType "value1"
+
+    // I guess, no way to support closing square brackets inside key names,
+    // thus this results in undefined.
+    type M = GetFieldType<O3, "['key[with]brackets']">; // $ExpectType "value2"
+    type N = GetFieldType<O3, '["key[with]brackets"]'>; // $ExpectType "value2"
+}
+
+{ // GetFieldType: additional tests for internals.
```

Comment:
I don't think it's worth testing these directly, it only makes things harder to maintain. As long as the top level GetFieldType tests work then we are good